rust/azure_iot_operations_services/src/azure_device_registry/client.rs

```diff
@@ -0,0 +1,484 @@
+    /// if there is an error returned by the ADR Service.
+    ///
+    /// [`struct@Error`] of kind [`AIOProtocolError`](ErrorKind::AIOProtocolError)
+    /// if there are any underlying errors from the AIO RPC protocol.
+    pub async fn notify_asset_endpoint_profile_update(
+        &self,
+        notification_type: bool,
+        timeout: Duration,
+    ) -> Result<NotificationResponse, AzureDeviceRegistryError> {
+        let payload = NotifyOnAssetEndpointProfileUpdateRequestPayload {
+            notification_request: if notification_type {
+                NotificationMessageType::On
+            } else {
+                NotificationMessageType::Off
+            },
+        };
+
+        let command_request = rpc_command::invoker::RequestBuilder::default()
+            .payload(payload)
+            .map_err(|e| AzureDeviceRegistryError(ErrorKind::SerializationError(e.to_string())))?
+            .timeout(timeout)
+            .build()
+            .map_err(|e| AzureDeviceRegistryError(ErrorKind::InvalidArgument(e.to_string())))?;
+
+        let result = self
+            .notify_on_asset_endpoint_profile_update_command_invoker
+            .invoke(command_request)
+            .await;
+
+        match result {
+            Ok(response) => Ok(response.payload.notification_response),
+            Err(e) => Err(AzureDeviceRegistryError(ErrorKind::from(
+                ErrorKind::AIOProtocolError(e),
+            ))),
+        }
+    }
+
+    /// Retrieves an asset from a Azure Device Registry service.
+    ///
+    /// # Arguments
+    /// * `asset_name` - The name of the asset.
+    /// * `timeout` - The duration until the Client stops waiting for a response to the request, it is rounded up to the nearest second.
+    ///
+    /// Returns a [`Asset`] if the the asset was found.
+    ///
+    /// # Errors
+    /// [`struct@Error`] of kind [`InvalidArgument`](ErrorKind::InvalidArgument)
+    /// if the `timeout` is zero or > `u32::max`, or there is an error building the request.
+    ///
+    /// [`struct@Error`] of kind [`SerializationError`](ErrorKind::SerializationError)
+    /// if there is an error serializing the request.
+    ///
+    /// [`struct@Error`] of kind [`ServiceError`](ErrorKind::ServiceError)
+    /// if there is an error returned by the ADR Service.
+    ///
+    /// [`struct@Error`] of kind [`AIOProtocolError`](ErrorKind::AIOProtocolError)
+    /// if there are any underlying errors from the AIO RPC protocol.
+    pub async fn get_asset(
+        &self,
+        asset_name: String,
+        timeout: Duration,
+    ) -> Result<Asset, AzureDeviceRegistryError> {
+        let get_request_payload = GetAssetRequestPayload {
+            asset_name: asset_name,
+        };
+
+        let command_request = rpc_command::invoker::RequestBuilder::default()
+            .payload(get_request_payload)
+            .map_err(|e| AzureDeviceRegistryError(ErrorKind::SerializationError(e.to_string())))?
```

Comment:
I think this can be simplified to a ? or an .into()? maybe. Let's take a look when we look at errors